### DISCUSSION TITLE
How to Modify Custom Variables for RITMs or Requested Items

### DISCUSSION THREAD

**Discussion by radcox** [Snow-Shell/servicenow-powershell#237] [General]
I am trying to update a custom variable field for an RITM, specifically a Termination Date Field and I am having trouble passing the correct Syntax to the object for the Hashtable entry.  

![image](https://user-images.githubusercontent.com/49378846/223164347-b2ea84f3-65f3-4600-b249-e81aad41f41f.png)

This data lives at $object.customvariable.termination_Date  of the object $object

I've tried this:

get-servicenowrecord -id $servicenowritmnumber -new -IncludeCustomVariable |  update-servicenowrecord  -values  @{'customvariable.termination_Date' = "$terminationeffectivedate"}

This part is obviously not right: 
![image](https://user-images.githubusercontent.com/49378846/223166810-88232900-6f28-46fb-8a16-7cb3217d0c16.png)

Any help on how to pass the hastable modification/entry is much appreciated.

**Comment by gdbarron** [comment]
CustomVariable is a property added by the module and not a table field.  There is a lot of work, and relevant tables, involved in 'getting' that data.  Some serious thought would need to be put into how to update it.  I have some ideas, but will take some time.

**Reply by radcox** [reply]
That makes sense, thanks for the update.

**Reply by gdbarron** [reply]
@radcox not sure if you saw, but I released this functionality in v4.0.
